lib/Sema/CodeSynthesis.cpp

```diff
@@ -321,6 +321,9 @@ static ConstructorDecl *createImplicitConstructor(NominalTypeDecl *decl,
            "Only 'distributed actor' type can gain implicit distributed actor init");
 
     if (swift::ensureDistributedModuleLoaded(decl)) {
+      // copy access level of distributed actor init from the nominal decl
+      accessLevel = decl->getEffectiveAccess();
```

Comment:
this fixes the accessibility of the created init to be the same as the class for a dist actor.